File: Alternating Current.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

	string str;
	cin >> str;
	stack<char> st;
	for (int i = 0; i < str.size(); i++) {
		if (st.empty()) {
			st.push(str[i]);
		} else {
			if (st.top() == str[i]) {
				st.pop();
			} else {
				st.push(str[i]);
			}
		}
	}
	(st.empty()) ? cout << "Yes\n" : cout << "No\n";
}

File: Alternating Current.py
s = input()
st = list()
for i in range(len(s)):
	if len(st) == 0:
		st.append(s[i])
	else:
		if st[len(st)-1] == s[i]:
			st.pop()
		else:
			st.append(s[i])
if len(st) == 0:
	print("Yes")
else:
	print("No")
